trunk/NDHMS/Routing/module_channel_routing.F

```diff
@@ -2058,100 +1872,109 @@ subroutine drive_CHANNEL_RSL(UDMP_OPT,KT, IXRT,JXRT,  &
             !! current would require global communication at the end of each loop through k
             !! (=kth reach). Additionally, how do you synchronize to make sure the upstream are all
             !! done before doing the downstream?
-            if(gQLINK(m,2) .gt. 0)   Quc = Quc + gQLINK(m,2)  !--accum of upstream inflow of current timestep (2)  
+            if(gQLINK(m,2) .gt. 0)   Quc = Quc + gQLINK(m,2)  !--accum of upstream inflow of current timestep (2)
             if(gQLINK(m,1) .gt. 0)   Qup = Qup + gQLINK(m,1)  !--accum of upstream inflow of previous timestep (1)
          end do ! do i
 #else
-         do m = 1, NLINKSL                
+         do m = 1, NLINKSL
             if (LINKID(k) .eq. TO_NODE(m)) then
                Quc = Quc + QLINK(m,2)  !--accum of upstream inflow of current timestep (2)
                Qup = Qup + QLINK(m,1)  !--accum of upstream inflow of previous timestep (1)
             endif
          end do ! do m
 #endif
       endif !note that we won't process type 2 links, since they are internal to a lake
-      
-      
+
+
       !yw ### process each link k,
       !       There is a situation that different k point to the same LAKEIDX
       !        if(TYPEL(k) .eq. 1 .and. LAKEIDX(k) .gt. 0) then   !--link is a reservoir
       if(TYPEL(k) .eq. 1 ) then   !--link is a reservoir
-         
+
          lakeid = LAKEIDX(k)
          if(lakeid .ge. 0) then
-            call LEVELPOOL(lakeid,Qup, Quc, tmpQLINK(k,2), &
-                 0.0, DTRT_CH, RESHT(lakeid), HRZAREA(lakeid), WEIRH(lakeid), LAKEMAXH(lakeid), &
-                 WEIRC(lakeid), WEIRL(lakeid),ORIFICEE(lakeid), ORIFICEC(lakeid), ORIFICEA(lakeid))
-            
+
+            ! Calls run_release for a single reservoir depending on its type as
+            ! in whether it uses level pool, machine learning, or persistence.
+            ! Inflow, lateral inflow, water elevation, and the routing period
+            ! are passed in. Updated outflow and water elevation are returned.
+            ! Domain ID (did) set to 1.
+            call rt_domain(1)%reservoirs(lakeid)%ptr%run_release(Qup, Quc, 0.0, &
```

Comment:
Magic number!?

Reply:
Changing to pass in 'did' from module_HYDRO_drv.F

Reply:
The number i was referencing was 0.0.

Reply:
That is for lateral inflow. It is passed as 0.0 also in master for RSL. Mehdi previously told me that link routing does not have lateral inflow.

Reply:
I'm against introducing more magic numbers like these. We need a module with numbers so they can be named and precised. That's a long-term, large-scale clean up effort. Just try to avoid this in the future. Since the arguments are not named (e.g. lateral_inflow), a comment here would also be helpful.